### ISSUE TITLE
Reconsider wrapping functions and unused custom hooks

### ISSUE THREAD

**Issue by iandunn** [imath/wordcamp-talks#36]
I'd like to discuss potentially removing the functions that wrap Core functions and hooks, because in my opinion the problems they introduce are worse than the problems they try to solve.

[`wct_user_can()`](https://github.com/imath/wordcamp-talks/blob/6cf7eef472effd434e9b81f93c0eabc4fd014913/includes/core/capabilities.php#L346-L356) is a good example. As far as I can tell, it doesn't add any significant functionality. The handling of `$args` doesn't seem to offer any benefit over calling `current_user_can()` directly, and the filter is never used by the plugin. If another plugin wanted to use the filter, it seems like they could just use the `has_cap` or `map_meta_cap` filters instead. Am I missing something?

These kinds of wrapping functions introduce several problems:
* They add extra and unnecessary lines of code to the plugin, which makes it more difficult and time-consuming to review and merge into WordCamp.org.
* They add extra points of failure. `current_user_can()` has been heavily tested for bugs, but a wrapper could unintentionally introduce a subtle bug. This is even more of a problem for an authorization function like `current_user_can()`, where the potential damage from a bug could be severe. I think there needs to be a compelling reason to add on to the Core API.
* Any developer who wants to maintain or contribute to the plugin has to learn a whole extra set of API calls, instead of just using the standardized Core API. 
* Adding custom filters for everything, instead of doing it in a more intentional and practical way, introduces a big backwards-compatibility burden, because any changes to the plugin have to consider how external plugins calling those filters will be affected. That leads to a lot of technical debt.

Another example is a function like `wct_register_taxonomies()`, which do nothing except call `do_action( 'wct_register_taxonomies' );`. Why can't the calling code just call `do_action()` directly?

Also, the functions like `wct_get_root_url()`, which do nothing except wrap a Core function with an unused filter.

There are entire files that are practically nothing but wrappers. Having all these wrappers ends up creating a maze for other developers to wade through whenever they want to work on something.

**Comment by tomjn** [comment]
I agree, and I've removed a number of wrappers already as I've come across them
